### PR TITLE
MAINT: adding sphinx 8.0.x to the testing matrix

### PR DESCRIPTION
**Pull request by bsipocz**

### CODE REVIEW

File: .github/workflows/python-tests.yml [hunk 1]
```diff
@@ -35,16 +35,19 @@ jobs:
...
             toxenv: py311-test-sphinxdev
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
       with:
         fetch-depth: 0
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
```
Comment (pllim) [inline .github/workflows/python-tests.yml:50]: Oh, maybe we should add dependabot too. If you don't want to do it in this PR, I can do it after merge.

File: tox.ini [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tox]
 envlist =
-    py{37,38,39,310,311}-test{-v2deps}{-sphinx30,-sphinx35,-sphinx40,-sphinx50,-sphinx60,-sphinx70,-sphinxdev}
+    py{37,38,39,310,311}-test{-v2deps}{-sphinx30,-sphinx35,-sphinx40,-sphinx50,-sphinx60,-sphinx70,-sphinx80-sphinxdev}
```
Comment (pllim) [inline tox.ini:3]:
```suggestion
    py{37,38,39,310,311}-test{-v2deps}{-sphinx30,-sphinx35,-sphinx40,-sphinx50,-sphinx60,-sphinx70,-sphinx80,-sphinxdev}
```

File: .github/workflows/python-tests.yml (hunk 1 above)
Comment (bsipocz) [reply .github/workflows/python-tests.yml:50]: done

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**